pkg/operator/encryption/preconditions.go

```diff
@@ -0,0 +1,106 @@
+package encryption
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/labels"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+
+	configv1informers "github.com/openshift/client-go/config/informers/externalversions/config/v1"
+	configv1listers "github.com/openshift/client-go/config/listers/config/v1"
+	"github.com/openshift/library-go/pkg/operator/encryption/encryptionconfig"
+	"github.com/openshift/library-go/pkg/operator/encryption/state"
+	operatorv1helpers "github.com/openshift/library-go/pkg/operator/v1helpers"
+)
+
+// PreconditionProvider determines if encryption controllers should synchronise.
+type PreconditionProvider interface {
```

Comment:
Java'esque name.

Interface with one func are odd. Use `type Precondition func() (bool, error)`. Why the extra layer of abstraction?